Tests/CorePromise/WhenTests.swift

```diff
@@ -5,22 +5,24 @@ import XCTest
 class WhenTests: XCTestCase {
 
     func testEmpty() {
-        let e1 = expectation(description: "")
         let promises: [Promise<Void>] = []
+        
+        let e1 = expectation(description: "")
         when(fulfilled: promises).done { _ in
             e1.fulfill()
         }.silenceWarning()
 
         let e2 = expectation(description: "")
         when(resolved: promises).done { _ in
             e2.fulfill()
-        }.silenceWarning()
+        }
```

Comment:
Does this mean the `when` has `@discardableResult` when it should not?

Reply:
It is not about `when`, but about `done`. 
`when(fulfilled:)` returns Promise and `done` for Thenable is declared **without** `@discardableResult`, however, `when(resolved:)` returns Guarantee and `done` for Guarantee is declared **with** `@discardableResult`.
That is why it does not need `.silenceWarning()`. It has **not** been changed in this PR.

Thenable:
https://github.com/mxcl/PromiseKit/blob/f8a593a320401045979e705640e654f6c2e0bba3/Sources/Thenable.swift#L167-L168

Guarantee:
https://github.com/mxcl/PromiseKit/blob/f8a593a320401045979e705640e654f6c2e0bba3/Sources/Guarantee.swift#L78-L80

Reply:
Oh, I'm a dumbo, thanks.